torchvision/prototype/transforms/functional/_geometry.py

```diff
@@ -1531,13 +1528,13 @@ def elastic_image_pil(
     return to_pil_image(output, mode=image.mode)
 
 
-def _create_identity_grid(size: Tuple[int, int], device: torch.device) -> torch.Tensor:
+def _create_identity_grid(size: Tuple[int, int], device: torch.device, dtype: torch.dtype) -> torch.Tensor:
```

Comment:
This somewhat violates the assumption in https://github.com/pytorch/vision/pull/7211/files#r1102883508, doesn't it? I could pass `dtype=torch.uint8` here and the function would not fail. Meaning it would give me a non-float grid.

Reply:
This is a private method, so should be called with uint8 and second, grid (sum of identity + displacement) still be float as displacement will be float

Reply:
I don't understand. This transform just creates a grid and it wouldn't fail if we pass in an integer dtype, right? I'm not saying this is a bad thing, but one of the assumptions you made above for `grid` in `_apply_grid_transform` is that it will always be a floating type. Maybe there always happens some computation in between, but I would prefer to not rely on this assumption there.

Reply:
Check the line 1510 in the PR. `grid` is constructed as `_create_identity_grid(...).add_(displacement)`. Actually, more exactly if we construct identity_grid with torch.uint8 and then call `.add_()` with floating displacement, the op will fail with `RuntimeError: result type Float can't be cast to the desired output type Byte`